### PR TITLE
Convert VCF to Fasta

### LINKED ISSUE #16: Convert VCFs to Fasta
We'll want to convert all of the VCFs to fasta format for the analysis of population recombination rate. Many programs (e.g., LDJump, LDhelmet) accept Fasta files as input. 

Since SLiM VCFs are all As and Ts, we'll sample a section of _Chlamydomonas_ genome of the same length as the genomic elements in SLiM. We'll then use a mutation matrix to overlay the substitution onto the chlamy reference sequence at positions specified in the VCF.

### LINKED ISSUE #17: Sort and Tabix index VCFs
We'll likely want the VCFs to be sorted and Tabix indexed for downstream analyses.

### LINKED ISSUE #18: Add Rproject and packrat to repo for R analyses
LDJump is an R packages and so we'll want to create an Rproject and push this to the repo. 

I still use packrat to manage my R dependencies but we could totally think of other ways to manage dependencies if there are better solutions (Packrat is sometimes a bit bloated). I think having some kind of dependency management system is essential though.

### PR DESCRIPTION
**Pull request by James-S-Santangelo**
Rather large PR that combines multiple issues. 

- Closes #16 
- Closes #17 
- Closes #18 

- Tabix index and sort VCFs exported from SLiM
- Use scripts previously written by Ahmed that converts the VCFs to Fasta
- Create Rproject for development of R scripts and use `Packrat` for R dependency management
- Some minor fixes to syntax, imports, requirements, etc.

### CODE REVIEW

File: scripts/python/SLiM_sims.py [hunk 1]
```diff
@@ -27,14 +32,49 @@ def run_slim(N, bot, outpath, slim_path):
...
-    # Find all VCFs in outpath
-    process_find = subprocess.Popen(['find', outpath, '-name', '*.vcf'],
+    # Use find utility to identify all VCFs in outpath
+    process_find = subprocess.Popen(['find', outpath, '-type', 'f',
+                                     '-name', '*.' + ext],
                                     stdout=subprocess.PIPE,
                                     stderr=subprocess.PIPE,
                                     universal_newlines=True)
 
+    return(process_find)
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: `return process_find`

File: scripts/python/SLiM_sims.py [hunk 2]
```diff
@@ -27,14 +32,49 @@ def run_slim(N, bot, outpath, slim_path):
...
+        out_name = outpath + filename + '_sorted.vcf'
+
+    # Files from 'find' are piped to 'xargs', which uses 'sort' to sort by position
+        process_sort = subprocess.Popen(['vcf-sort', vcf],
+                                        # stdin=process_find.stdout,
+                                        stdout=open(out_name, 'w'),
+                                        stderr=subprocess.PIPE,
+                                        universal_newlines=True)
+
+        out, err = process_sort.communicate()
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: Potentially don't need to unpack `out` here - could be worth looking into this

File: scripts/python/SLiM_sims.py [hunk 3]
```diff
@@ -47,25 +87,72 @@ def bgzip(outpath):
     print(err)
 
 
+def tabix_vcfs(outpath):
+
+    # Use find utility to identify all VCFs in outpath
+    process_find = find_vcfs(outpath, 'vcf.gz')
+
+    # bgzip all found VCFs
+    process_tabix = subprocess.Popen(['xargs', '-n1', 'tabix', '-f'],
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: Could probably add `-p vcf` to be explicit about tabix format

File: scripts/python/SLiM_sims.py [hunk 4]
```diff
@@ -47,25 +87,72 @@ def bgzip(outpath):
...
+def vcf2fasta(table, region, mut_mat, outpath):
+
+    fasta_outpath = outpath + "fasta-files/"
+    create_output_directory(fasta_outpath)
+
+    for vcf in tqdm(glob(outpath + '*.vcf.gz')):
+
+        filename = fasta_outpath + vcf.split('/')[-1].split('.vcf.gz')[0] + ".fasta"
+
+        slim_vcf2fasta_chrom.write_fasta(vcf, table, region, mut_mat, filename)
```
Comment (aays) [inline scripts/python/SLiM_sims.py:130]: Leaving note here from in-person meeting - look for ways to incorporate `multiprocessing` with chunks of filenames to speed up vcf -> fasta processing

File: scripts/python/SLiM_sims.py [hunk 5]
```diff
@@ -47,25 +87,72 @@ def bgzip(outpath):
...
-    N = args.N
+    N = args.pop_size
     bot = args.bot
     slim_path = args.slim_path
+    table = args.table
+    region = args.region
+    mut_mat = args.mut_mat
     outpath = str(args.outpath) + "N{0}_bot{1}/".format(N, bot)
 
     # Create output directory, if it doesn't exit
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: s/exit/exist

File: scripts/python/SLiM_sims.py [hunk 6]
```diff
@@ -47,25 +87,72 @@ def bgzip(outpath):
...
     create_output_directory(outpath)
 
     # Run simulations
-    run_slim(N, bot, outpath, slim_path)
+    # run_slim(N, bot, outpath, slim_path)
+
+    # Sort VCFs
+    # sort_vcfs(outpath)
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: Remember to uncomment!

File: scripts/python/SLiM_sims.py [hunk 7]
```diff
@@ -47,25 +87,72 @@ def bgzip(outpath):
...
 
     # bgzip files
-    bgzip(outpath)
+    # bgzip_vcfs(outpath)
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: Add `print` statements in the functions for progress info

File: scripts/python/SLiM_sims.py [hunk 8]
```diff
@@ -27,14 +32,49 @@ def run_slim(N, bot, outpath, slim_path):
...
 
+    return(process_find)
+
+
+def sort_vcfs(outpath):
+
+    # Use find utility to identify all VCFs in outpath
+    # process_find = find_vcfs(outpath, 'vcf')
+
+    for vcf in glob(outpath + '*.vcf'):
```
Comment (aays) [inline scripts/python/SLiM_sims.py]: Add `tqdm`, progress messages, and counter here

File: scripts/python/slim_vcf2fasta_chrom.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+
+def args():
+    parser = argparse.ArgumentParser(
+        description='slim vcf -> fasta w/ reference',
+        usage='python3.5 slim_vcf2fasta_chrom.py [options]')
+
+    parser.add_argument('-v', '--vcf', required=True,
+                        type=str, help='bgzipped + tabixed vcf')
+    parser.add_argument('-t', '--table', required=True,
```
Comment (aays) [inline scripts/python/slim_vcf2fasta_chrom.py:24]: Low-priority enhancement - remove annotation table here, and have file read in a text file containing just the sequence in fasta format (single sequence, header = region, len(sequence) = 1e6)

Could have separate script that actually queries annotation table and creates sequence, while this script just reads in that text file

**Comment by James-S-Santangelo** [comment]
Shadow merging since no additional review required.

**Pull request merged**